weaknesses/MASVS-RESILIENCE/MASWE-0098.md

```diff
@@ -2,17 +2,43 @@
+
+Dynamic instrumentation tools, like [GameGuardian](https://gameguardian.net/forum/topic/20322-gameguardian-work-without-root/), can leverage Android virtualization software to operate on non-rooted devices.
+
+## Impact
+
+Maliciously used Android virtualization software can lead to a variety of security issues, including but not limited to:
+
+- **Unauthorized reading of application private data**: Malware utilizing Android virtualization software to virtualize a target application can access its private data. This can lead to exposure of sensitive user information, such as login credentials, financial data, and personal messages, compromising user privacy and security.
+- **Unauthorized alteration of application private data**: By injecting malicious code through virtualization, malware can modify private data within an application. This alteration can result in unauthorized transactions, changes to user settings, or tampering with the application's functionality, potentially causing financial loss or operational disruptions.
+- **Unauthorized escalation of privileges or identity theft**: Virtualization software can create a more permissive environment, enabling the execution of other malicious tools that would normally require root permissions.
+- **Circumventing security mechanisms**: Virtualizers can dynamically tamper with the application's memory without altering the application's integrity. This allows malicious code to bypass certain anti-tampering security measures.
+
+## Modes of Introduction
+
+Virtualization software introduces an additional layer between an application and its underlying operating environment, fundamentally altering how applications interact with system resources. This extra layer creates a more permissive environment, which can have significant implications for both legitimate use and potential threats.
+
+In a virtualized environment, the software effectively isolates the application from the actual operating system and hardware, allowing the virtualization software to monitor and control any interactions with the system. This is normally not possible on a non-rooted system.
+
+For example, by emulating or partially recreating the Android system within an application sandbox, virtualization software can fully interact with the virtualized environment, even though the app appears to be running on a non-modified Android system. The virtualization software has full control over what happens inside of the virtualized system and is not restricted by typical security constraints, which makes it easier to analyze, manipulate, or modify the target app's behavior.
+
+## Mitigations
+
+To mitigate the risks associated with Android virtualization software, developers should:
```

Comment:
Add something like "Opt out of on-device Android container apps via the REQUIRE_SECURE_ENV manifest flag"